agent/command.go

```diff
@@ -114,6 +115,33 @@ func (a *Agent) runCommandSet(ctx context.Context, tc *taskContext, commandInfo
 			tc.setCurrentIdleTimeout(nil)
 		}
 
+		if cmd.Name() == "manifest.load" {
+			manifestLoadCalled = true
+		}
+
+		if cmd.Name() == "git.get_project" && !manifestLoadCalled {
+			tc.logger.Task().Infof("calling manifest.load on variant %s ",
+				tc.taskConfig.BuildVariant.Name)
+			a.runCommands(ctx, tc, []model.PluginCommandConf{
```

Comment:
Ideally we would run this as _part_ of git.get_project as opposed to it's own command here. Can git.get_project have a new parameter "runManifestLoad" or something that can be added to the params? And the code in manifest.load could be moved to a helper function that can be called explicitly by git.get_project? I'm thinking a future outcome of this work would be to remove the manifest.load command.

Reply:
Based on what Ryan explained [here](https://jira.mongodb.org/browse/EVG-12839), it doesn't seem like it would be useful to take it in as a parameter and let users set it to true or false because git.get_project should never run without manifest.load.

Reply:
The problem with this approach would also be that we don't know if manifest.load was already called, and we may end up repeating work a lot of the time.

Reply:
IMHO this change is 👍🏽 as it is. We could shuffle things around to make the manifest.load logic a true part of get_project and make manifest.load a true nop and remove/nop/deprecate the command but that feels like a future optimization.

:shipit::shipit::shipit:

Reply:
What I meant was that _we_ could add manifestLoadCalled to commandInfo.Params (rather than asking users to do it) and then at the top of git.get_project we could call the manifest load logic. I'm happy to pair on this if it isn't clear.

I'd really like to avoid putting this much command specific implementation logic in the agent, which is already pretty intricate. If we're trying to change the behavior of the git.get_project it seems like it belongs in the command itself.

Reply:
I deprecated the manifest.load command and moved the logic into git.get_project.